warehouse/query-core/src/main/java/datawave/query/attributes/Document.java

```diff
@@ -280,77 +281,110 @@ public void put(String key, Attribute<?> value, Boolean includeGroupingContext,
                 
                 // When calling put() on a Document which already contains an Attributes
                 // for a given with another Attributes, issue the equivalent of a putAll() on the new Attributes
-                // to not create additional hiearchy inside this Document
+                // to not create additional hierarchy inside this Document
                 //
                 // e.g. Given: {"CONTENT"=>[BODY:foo, HEAD:foo]}.put("CONTENT", [BODY:bar, HEAD:bar, FOOT:bar])
                 // We want to get: {"CONTENT"=>[BODY:foo, HEAD:foo, BODY:bar, HEAD:bar, FOOT:bar]}
                 // *not*: {"CONTENT"=>[BODY:foo, HEAD:foo, [BODY:bar, HEAD:bar, FOOT:bar]]}
                 
                 if (value instanceof Attributes && existingAttr instanceof Attributes) {
-                    // merge the two sets
-                    attrs = (Attributes) existingAttr;
-                    
-                    _count -= attrs.size();
-                    if (trackSizes) {
-                        _bytes -= attrs.sizeInBytes();
-                    }
-                    
-                    attrs.addAll(((Attributes) value).getAttributes());
-                    
-                    _count += attrs.size();
-                    if (trackSizes) {
-                        _bytes += attrs.sizeInBytes();
+                    // ensure there are no fuzzy matches before merging
+                    if (!AttributeComparator.multipleToMultiple((Attributes) existingAttr, (Attributes) value)) {
+                        // merge the two sets
+                        attrs = (Attributes) existingAttr;
+                        
+                        _count -= attrs.size();
+                        if (trackSizes) {
+                            _bytes -= attrs.sizeInBytes();
+                        }
+                        
+                        attrs.addAll(((Attributes) value).getAttributes());
+                        
+                        _count += attrs.size();
+                        if (trackSizes) {
+                            _bytes += attrs.sizeInBytes();
+                        }
+                    } else {
+                        // fuzzy matches found, attempt to combine attributes
+                        Set<Attribute<? extends Comparable<?>>> combinedSet = AttributeComparator.combineMultipleAttributes((Attributes) existingAttr,
+                                        (Attributes) value);
+                        Attributes mergedAttributes = new Attributes(combinedSet, this.isToKeep(), trackSizes);
+                        dict.put(key, mergedAttributes);
```

Comment:
`_count` and `_bytes` is not being updated

Reply:
Noted and updated